.github/workflows/run-tests.yml

```diff
@@ -12,119 +12,138 @@ jobs:
   compile:
     runs-on: ubuntu-latest
     timeout-minutes: 90
+    container:
+      image: ghcr.io/spinalhdl/docker:master
     steps:
     - uses: actions/checkout@v3
-    - uses: ./.github/actions/sbt
+    - run: git config --global --add safe.directory $GITHUB_WORKSPACE
```

Comment:
I have created a PR at the SpinalHDL/docker#6 which can make sure add safe.direcotry as default.

On other hand, the config of safe.directory is not only for submodules, there are cases that compile fail for the main stream code.

Reply:
After thinking about it: it depends on the user that is run, if you do a `git config --global` then it's for all repositories of the current user. To set it consistently for all users it would have to be `git config --system`

Reply:
I think this is done.